### ISSUE TITLE
Windows public api declaration seems broken.

### ISSUE THREAD

**Issue by mgautierfr** [openzim/libzim#808]
The last change made in https://github.com/openzim/libzim/pull/796 seems broken.
The PR #807 add some ddl export of forgotten symbol. 

But it seems it brokes something on the compilation of kiwix-tools.
Now we have a lot (if not all) of libkiwix symbols not found. I don't know why.

@xiaoyifang can you try to compile kiwix-tools on windows ? Or do a cross-compilation to win32 if you are on linux (`kiwix-build kiwix-tools --target-platform win32_dyn`) ?

**Comment by xiaoyifang** [comment]
https://github.com/openzim/libzim/blob/d87ff0d0d80d251ce976d34bac96ffc53c04bfe4/include/zim/zim.h#L38

should `|| defined __CYGWIN__` be removed ,if it can work before?

**Comment by mgautierfr** [comment]
Setting `LIBZIM_API` only if we compile with Ms Compiler seems to work (at least on Cygwin).
@xiaoyifang can you test #807 on Windows to be sure it doesn't brake on Windows ?

**Comment by xiaoyifang** [comment]
Ok ,I'll let you know when I tested

**Comment by xiaoyifang** [comment]
https://github.com/xiaoyifang/vcpkg/blob/3880056be47c74f1ebe709c8671062ad18ef2292/ports/libzim/cross-builds.diff

This is what vcpkg community handled this.

**Comment by xiaoyifang** [comment]
> #if defined(_MSC_VER) && defined(LIBZIM_EXPORT_DLL)

can work on windows.

But I would recommend changes like this

https://github.com/microsoft/vcpkg/blob/3880056be47c74f1ebe709c8671062ad18ef2292/ports/libzim/dllexport.diff#L33-L36

such as 
```
 
+if host_machine.system() == 'windows' and get_option('default_library') == 'shared'
+  public_conf.set('LIBZIM_EXPORT_DLL', true)
+endif
+
```

**Comment by kelson42** [comment]
@mgautierfr Can we foreseen to fix this today:
* This breaks the CI in various repos
* This is the last blocker for `8.2.1` release

**Comment by mgautierfr** [comment]
@xiaoyifang  I will merge #959 anyway as it fix our current problem.

I'm not sure about your solution, as it will set `decl(export)` on gcc cross-compilation to windows and we don't want that.

**Comment by kelson42** [comment]
@mgautierfr What's up here? We need to release `8.2.1` ASAP.

**Comment by mgautierfr** [comment]
This is fixed by #807
